dbt/adapters/spark/impl.py

```diff
@@ -132,38 +134,79 @@ def add_schema_to_cache(self, schema) -> str:
         # so jinja doesn't render things
         return ""
 
+    def parse_information(self, table_name):
+        information = ""
+        kwargs = {"table_name": table_name}
+        try:
+            table_results = self.execute_macro(DESCRIBE_TABLE_EXTENDED_MACRO_NAME, kwargs=kwargs)
+        except dbt.exceptions.RuntimeException as e:
+            logger.debug(f"Error while retrieving information about {table_name}: {e.msg}")
+            return []
+        for info_row in table_results:
+            info_type, info_value, _ = info_row
+            if info_type.startswith("#") is False:
+                information += f"{info_type}: {info_value}\n"
+        return information
+
     def list_relations_without_caching(
         self, schema_relation: SparkRelation
     ) -> List[SparkRelation]:
         kwargs = {"schema_relation": schema_relation}
+        try_show_tables = False
+        expected_result_rows = 4
         try:
             results = self.execute_macro(LIST_RELATIONS_MACRO_NAME, kwargs=kwargs)
         except dbt.exceptions.RuntimeException as e:
```

Comment:
It may be true that we don't have more granular exceptions than RuntimeException. But I want to make sure we don't before parsing the error message to determine the type of error. I'd rather submit a PR to `dbt-core` before parsing error messages for the error type.

Reply:
That's a good point. I just double-checked and there is no more granular exception thrown. This pattern is already used in the code, would it be possible to follow up on this in a separate PR? WDYT?